app/code/Magento/Checkout/Model/GuestPaymentInformationManagement.php

```diff
@@ -139,9 +140,13 @@ public function savePaymentInformationAndPlaceOrder(
         }
         try {
             $orderId = $this->cartManagement->placeOrder($cartId);
-        } catch (\Magento\Framework\Exception\LocalizedException $e) {
+        } catch (LocalizedException $e) {
             $this->logger->critical(
-                'Placing an order with quote_id ' . $cartId . ' is failed: ' . $e->getMessage()
+                'Placing an Order failed (reason: '.  $e->getMessage() .')',
```

Comment:
I think "reason" might be moved to "context" too. Also, we could add boolean "is_guest_checkout" to the context as well.